### ISSUE TITLE
cannot import name '_compare_version' from 'torchmetrics.utilities.imports' (C:\Code\warpfusion\v5.27.5\env\lib\site-packages\torchmetrics\utilities\imports.py)

### ISSUE THREAD

**Issue by Begadd** [Sxela/WarpFusion#61]
any help ?

**Comment by nimnim111** [comment]
try
pip install torchmetrics==0.11.4

**Comment by Sxela** [comment]
If you're getting this: 
ImportError: cannot import name '_compare_version' from 'torchmetrics.utilities.imports' (/usr/local/lib/python3.10/dist-packages/torchmetrics/utilities/imports.py)


Fix for torchmetrics available:
v0.16: https://www.patreon.com/posts/84838689
v0.13: https://www.patreon.com/posts/stable-v0-13-new-83087458
kudos to @tomato
(and in newer versions)

For other versions: add 
!python -m pip -q install torchmetrics==0.11.4
 in a cell below 1.3 -> restart + run all

**Comment by 808vita** [comment]
I faced this error as well error as well. This was caused by two different files.
"env\lib\site-packages\torchmetrics\utilities\imports.py"

"wrapfusion-v0.8.6\env\lib\site-packages\pytorch_lightning\callbacks\progress\rich_progress.py"

I tried to downgrade but did not fix things .
I have edited the package files directly to fix compare_version method errors 
replaced all "_compare_version" in those two files into "compare_version"  (without underscore at the beginning).
and that worked for me . (but not the best way)


note: 
-try out low resolution first , did work on 4gb gpu; 
-16gb gpu is the recommended soec oof.
-needs cuda toolkit to be installed 
-requires 15+gb of storage space for packages , models etc
